ui/raidboss/data/03-hw/dungeon/gubal_library_hard.js

```diff
@@ -0,0 +1,300 @@
+        if (data.fluid[data.me] == data.fluid['Liquid Flame']) {
+          return {
+            en: 'Close to boss',
+          };
+        }
+        return {
+          en: 'Away from boss',
+        };
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid Cleanup',
+      regex: / 1B:\y{ObjectId}:\y{Name}:....:....:003[01]:/,
+      delaySeconds: 5,
+      run: function(data) {
+        delete data.fluid;
+      },
+    },
+    {
+      id: 'Gubal Hard Slosh',
+      regex: / 23:\y{ObjectId}:Liquid Flame:\y{ObjectId}:(\y{Name}):....:....:0039/,
+      condition: function(data, matches) {
+        return data.me == matches[1];
+      },
+      infoText: {
+        en: 'Away from boss',
+      },
+    },
+    {
+      id: 'Gubal Hard Seals',
+      regex: / 1A:\y{ObjectId}:(\y{Name}) gains the effect of (Moon|Sun)seal/,
+      condition: function(data, matches) {
+        return data.me == matches[1];
+      },
+      infoText: function(data, matches) {
+        if (matches[2] == 'Moon') {
+          return {
+            en: 'Stand in blue',
+          };
+        }
+        return {
+          en: 'Stand in red',
+        };
+      },
+    },
+    {
+      // This inflicts a vulnerability stack on the tank if not interrupted
+      id: 'Gubal Hard Condensed Libra',
+      regex: / 14:198D:Mechanoscribe starts using Condensed Libra on \y{Name}/,
+      infoText: function(data) {
+        if (data.CanStun()) {
+          return {
+            en: 'Stun Mechanoscribe',
+          };
+        }
+        if (data.CanSilence()) {
+          return {
+            en: 'Interrupt Mechanoscribe',
+          };
+        }
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Quakes',
+      regex: / 14:1956:Strix starts using On The Properties Of Quakes/,
+      infoText: {
+        en: 'Stand in light circle',
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Tornadoes',
+      regex: / 14:1957:Strix starts using On The Properties Of Tornados/,
+      infoText: {
+        en: 'Stand in dark circle',
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Imps',
+      regex: / 14:1959:Strix starts using On The Properties Of Imps/,
+      infoText: {
+        en: 'Cleanse in green circle',
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Thunder',
+      regex: /14:195A:Strix starts using On The Properties Of Thunder III/,
+      infoText: {
+        en: 'Spread',
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Darkness II',
+      regex: / 14:1955:Strix starts using On The Properties Of Darkness II/,
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      infoText: {
+        en: 'AoE',
+      },
+    },
+    {
+      id: 'Gubal Hard Ecliptic Meteor',
+      regex: /14:195D:Behemoth Ward starts using Ecliptic Meteor/,
+      delaySeconds: 14, // Leaving about 10s warning to complete the LoS
+      alertText: {
+        en: 'LoS Behemoth with boulder',
```

Comment:
Maybe just `hide behind boulder`, capitalized however you prefer?

Reply:
You're right, *what* over *how*. The meteor falling is obvious enough.